### PR TITLE
Use scala.reflect.TypeTest in Reflection

### PR DESCRIPTION
**Pull request by nicolasstucki**

### CODE REVIEW

File: scala3doc/src/dotty/dokka/tasty/TypesSupport.scala [hunk 1]
```diff
@@ -242,7 +242,8 @@ trait TypesSupport:
         }
         inner(sc) ++ texts(" match {\n") ++ casesTexts ++ texts("}")
 
-      case TypeIdent(t) => texts(t)
+      // TODO: Unreachable. TypeIdent is a TypeTree and not a TypeRep. Should this be cover in antother way?
+      // case TypeIdent(t) => texts(t)
```
Comment (nicolasstucki) [inline scala3doc/src/dotty/dokka/tasty/TypesSupport.scala:246]: @abgruszecki, the new TypeTest did its job and found this unreachable case. Should this be just removed or is it a potential bug?

Comment (abgruszecki) [reply scala3doc/src/dotty/dokka/tasty/TypesSupport.scala:246]: I'd remove it and see what the CI says.

Comment (nicolasstucki) [reply scala3doc/src/dotty/dokka/tasty/TypesSupport.scala:246]: The CI should pass, this case never matches at runtime.

Comment (abgruszecki) [reply scala3doc/src/dotty/dokka/tasty/TypesSupport.scala:246]: That's good enough for me, just remove it then.

File: compiler/src/scala/quoted/internal/impl/QuoteContextImpl.scala [hunk 2]
```diff
@@ -127,9 +126,8 @@ class QuoteContextImpl private (ctx: Context) extends QuoteContext, QuoteUnpickl
     type PackageClause = tpd.PackageDef
 
     object PackageClauseTypeTest extends TypeTest[Tree, PackageClause]:
-      def runtimeClass: Class[?] = classOf[PackageClause]
-      override def unapply(x: Any): Option[PackageClause] = x match
-        case x: tpd.PackageDef @unchecked => Some(x)
+      def unapply(x: Tree): Option[PackageClause & x.type] = x match
+        case x: (tpd.PackageDef & x.type) => Some(x)
```
Comment (liufengyun) [inline compiler/src/scala/quoted/internal/impl/QuoteContextImpl.scala:130]: Digression: it seems to be a good case for flow-sensitive `isInstanceOf`.

**Review by liufengyun — APPROVED** [review]
LGTM

**Pull request merged**